Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp

```diff
@@ -141,8 +247,37 @@ evdevDevice::~evdevDevice()
   }
 }
 
+void evdevDevice::ChangeFd(int new_fd)
+{
+  std::lock_guard<std::mutex> lk(m_device_mutex);
+  if (!m_initialized)
+    return;
+  int ret = libevdev_change_fd(m_dev, new_fd);
+  if (ret != 0)
+    ERROR_LOG(SERIALINTERFACE, "error changing fd to %d", new_fd);
+}
+
+bool evdevDevice::IsStillValid() const
+{
+  int current_fd = libevdev_get_fd(m_dev);
```

Comment:
I'm not sure if libevdev devices are thread-safe. Logically there shouldn't be any race condition because UpdateInput() will just use the old file descriptor before ChangeFd() completes. 

However, I added the mutex after I had Dolphin freeze once, and gdb revealed that it was blocking on a read() call somewhere in libevdev during an input update.

I was never able to reproduce the issue again though, so it's unsure if this is needed, or even if the issue was actually related to this PR.